report/2.Analysis/Scope.tex

```diff
@@ -0,0 +1,43 @@
+%This section is meant to specify the scope of project, both in terms of the report, the language that will be developed and the compiler.
+\section{Scope}
+As stated in the above analysis, the purpose of this project is to design, create and implement a programming language for the Arduino platform that is simple for beginners to use, whilst still teaching the fundamentals of programming.
+The language is meant to enable beginners to quickly and easily write
+code that can be executed, without losing the functionality of a high-level language.
+
+It is essential to consider the needed functionality in order to responsibly compose a language for beginners.
+After conducting interviews with gymnasium students, who can be described as beginners at programming, it was decided that the language for this project should be developed as an introductory language for beginners to use, before making a switch to a fully fleshed out programming language.
+
+As such, the language is meant to introduce the thought process used in programming, while ensuring that the user will not have to worry too much about syntax.
+%insert a reference on the next line to the ArduinoInEducation section
+As was discovered during the visit to the gymnasium class (section \ref{htxinter}), there were a couple of concepts that would be especially important to be wary of when developing the language.
+
+The students seemed to have issues with general syntax, data types, defining functions, control-structures and operator precedence.
+Therefore the developed language will need to have a heightened focus on these particular concepts. 
+
+Adding to these requirements, the project group would also like to reduce the time spent on program-setup as much as possible, allowing the users of the language developed to quickly get from idea to working solution.
+The goal for the language should always be simplicity, comprehensibility and ease of use.
+
+As the group intends to develop an "English-like" language, in the sense that the syntax should look very much like an English sentence, it has been concluded by the group, that almost all operations in the language will be a function. 
```

Comment:
this maybe shouldn't be in discussion. not sure - but function decisions is more design than analysis.

Reply:
however the "on top of this" sentence makes sense. but saying that everything should be functions (not almost.. but "all") should probably be in design

Reply:
I think it is fine here. But I see your point. Further argument for this decision should be in the design documentation.

Reply:
That is being done in language specification, seen as solved.